src/main/java/at/hannibal2/skyhanni/features/inventory/ItemPickupLog.kt

```diff
@@ -0,0 +1,314 @@
+package at.hannibal2.skyhanni.features.inventory
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.events.PurseChangeEvent
+import at.hannibal2.skyhanni.events.SackChangeEvent
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.CollectionUtils.addItemStack
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.ItemNameResolver
+import at.hannibal2.skyhanni.utils.ItemUtils.getInternalNameOrNull
+import at.hannibal2.skyhanni.utils.ItemUtils.getItemRarityOrNull
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.NEUInternalName.Companion.asInternalName
+import at.hannibal2.skyhanni.utils.NEUItems.getItemStack
+import at.hannibal2.skyhanni.utils.NumberUtil.addSeparators
+import at.hannibal2.skyhanni.utils.NumberUtil.shortFormat
+import at.hannibal2.skyhanni.utils.RenderUtils.renderRenderable
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockItemModifierUtils.getExtraAttributes
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraft.client.Minecraft
+import net.minecraft.item.ItemStack
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import java.util.Objects
+import kotlin.math.absoluteValue
+import kotlin.time.Duration.Companion.seconds
+
+@SkyHanniModule
+object ItemPickupLog {
+
+    enum class DisplayLayout(private val display: String) {
+        ICON("§e✎"),
+        ITEM_NAME("§d[Cute] Skirtwearer's Cake Soul"),
+        CHANGE_AMOUNT("§a+256"),
+        ;
+
+        override fun toString() = display
+    }
+
+    private val config get() = SkyHanniMod.feature.inventory.itemPickupLogConfig
+    private val coinIcon = "COIN_TALISMAN".asInternalName()
+    private var itemList = mutableMapOf<Int, Pair<ItemStack, Int>>()
```

Comment:
use PrimitiveItemStack as value

Reply:
PrimitiveItemStack doesnt work here since the name of the item comes directly from the internal name which has the same issue we talked about yesterday with the ItemAddInInventoryEvent
![image](https://github.com/user-attachments/assets/2d0f8abf-ca89-4ce7-9f91-000a05c36553)

Reply:
fair.